packages/grafana-ui/src/components/DataLinks/DataLinkInput.tsx

```diff
@@ -81,6 +82,12 @@ export const DataLinkInput: React.FC<DataLinkInputProps> = memo(
     const stateRef = useRef({ showingSuggestions, suggestions, suggestionsIndex, linkUrl, onChange });
     stateRef.current = { showingSuggestions, suggestions, suggestionsIndex, linkUrl, onChange };
 
+    // Used to get the height of the suggestion elements in order to scroll to them.
+    const selectedRef = useRef<HTMLDivElement>(null);
+    function getScrollTop() {
+      return (selectedRef.current?.clientHeight ?? 0) * suggestionsIndex;
+    }
```

Comment:
Maybe use `useMemo` here instead and use the `scrollTop` value in CustomScrollbar? I mention this because it seems like everything in this component is optimized.

```suggestion
    const scrollTop = useMemo(() => (selectedRef.current?.clientHeight ?? 0) * suggestionsIndex, [
      selectedRef,
      suggestionsIndex,
    ]);
```

Reply:
Using ref as a hook dependency is an antipattern (read more here https://medium.com/@teh_builder/ref-objects-inside-useeffect-hooks-eb7c15198780).

What you can do instead, to avoid this function creation is to declare it outside of the component scope and make it accept a dom element as an argument;

```
function getScrollTop(el, suggestionsIndex) {
  return (el.clientHeight ?? 0) * suggestionsIndex;
  }
```

Then the scrollTop will invalidate only on index change (which I think is sufficient):

```
const scrollTop = useMemo(() => getScrollTop(selectedRef.current, suggestionsIndex), [
	      suggestionsIndex,
	    ]);

Reply:
Aha, outstanding catch @dprokop 🎉

Reply:
implemented, also improved naming a bit (subjective)